homeassistant/components/ziggo_next/media_player.py

```diff
@@ -0,0 +1,183 @@
+"""Support for interface with a Ziggo Mediabox Next."""
+import logging
+import random
+
+from ziggonext import ONLINE_RUNNING, ONLINE_STANDBY, ZiggoNext
+
+from homeassistant.components.media_player import MediaPlayerDevice
+from homeassistant.components.media_player.const import (
+    MEDIA_TYPE_APP,
+    MEDIA_TYPE_CHANNEL,
+    SUPPORT_NEXT_TRACK,
+    SUPPORT_PAUSE,
+    SUPPORT_PLAY,
+    SUPPORT_PREVIOUS_TRACK,
+    SUPPORT_SELECT_SOURCE,
+    SUPPORT_TURN_OFF,
+    SUPPORT_TURN_ON,
+)
+from homeassistant.const import (
+    STATE_OFF,
+    STATE_PAUSED,
+    STATE_PLAYING,
+    STATE_UNAVAILABLE,
+)
+
+from .const import DOMAIN, ZIGGO_API
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_devices):
+    """Set up entry."""
+    players = []
+    api = hass.data[ZIGGO_API]
+    for box_id in api.settopBoxes.keys():
+        box = api.settopBoxes[box_id]
+        players.append(ZiggoNextMediaPlayer(box_id, box.name, api))
+    async_add_devices(players, update_before_add=True)
+
+
+class ZiggoNextMediaPlayer(MediaPlayerDevice):
+    """The home assistant media player."""
+
+    @property
+    def device_info(self):
+        """Return device info."""
+        return {
+            "identifiers": {
+                # Serial numbers are unique identifiers within a specific domain
+                (DOMAIN, self.box_id)
+            },
+            "name": self.box_name,
+            "manufacturer": "Ziggo",
+            "model": "Mediabox Next",
+        }
+
+    def __init__(self, box_id: str, name: str, api: ZiggoNext):
+        """Init the media player."""
+        self.api = api
+        self.box_id = box_id
+        self.box_name = name
+        self.box_state = None
+        self.box_info = None
+
+    def update(self):
+        """Update the box."""
+        self.api.load_channels()
+        box = self.api.settopBoxes[self.box_id]
```

Comment:
You could set `self.box` (or `self._box` to mark it as internal) in the constructor and use that here.